woocommerce/woocommerce-ios
# [Mobile Payments] Update LearnMore link in IPP flows

Closes: #6045 


### Description


Implements navigation to IPP documentation based on the active payment gateway plugin - wcpay/stripe. If the active plugin is `Stripe` the app navigates to `https://docs.woocommerce.com/document/stripe/accept-in-person-payments-with-stripe/`, if it's `wcpay` or unknown it navigates to `https://docs.woocommerce.com/document/getting-started-with-in-person-payments-with-woocommerce-payments/`.

The app supports navigation to IPP documentation from - onboarding error screens (`InPersonPaymentsViewController`) and connect to reader screen (`CardReaderSettingsSearchingViewController`).

I've tried different approaches and opted for introducing an action in Payments Store which returns the currently active plugin. The `CardReaderSettingsSearchingViewModel` can invoke this action and choose the URL based on the active plugin.

SideNotes:
- I also tried using `async/await` property and it worked nicely, however, the CI was complaining that async properties are not supported on ios 14 and lower.
- I also tried passing the active plugin into viewmodel's/controller's constructors and adding "docsUrl(..)" method in `CardPresentPaymentsConfiguration` but it turned out to be tricky, since we are using some kind of parent VM/Controller (CardReaderSettingsViewModelsOrderedList) and I couldn't figure out how to pass the active plugin from `CardReaderSettingsPresentingView` and it felt more hacky than the current solution anyway.
- I also considered returning a configuration instead of the active plugin directly from the newly introduced action, but since we already have `CardPresentPaymentsConfiguration` which doesn't know about the active plugin, introducing yet another configuration felt weird.

I think it might be worth to consider storing the active plugin into a persistent key-value store, but that's above the scope of this PR. Wdyt?

### Testing instructions


1. Select a store with stripe extension installed but which fails the onboarding checks - eg. outdated plugin version
2. Select "In Person Payments" in the app settings
3. Tap on "learn more" button and notice you are navigated to "IPP with Stripe" documentation
4. Repeat the same steps with wcpay extension and verify "IPP with wcpay" documentation is shown
-------------------
1. Select a store with stripe extension installed and onboarding completed
2. Select "In Person Payments" in the app settings
3. Tap on "Manage card reader"
4. Tap on "Learn more" and notice you are navigated to "IPP with Stripe" documentation
5. Repeat the same steps with wcpay extension and verify "IPP with wcpay" documentation is shown

### Screenshots



https://user-images.githubusercontent.com/2261188/157836452-51e9df3e-0fff-4c00-b7f7-06d2ab543392.mov


https://user-images.githubusercontent.com/2261188/157837207-51eb6672-41d5-4a76-93fe-0d3d3c9511ab.mov




---
